app/views/labware/_asset_summary.html.erb

```diff
@@ -100,5 +100,23 @@
       <td><%= asset.storage_location %></td>
     </tr>
   <% end %>
+  <tr>
+    <th>Retention Instruction</th>
+    <!-- First, check if retention instructions are there. -->
+    <% if asset.retention_instruction.blank? %>
+      <!-- If retention instructions aren't there in the labware table, check whether if the labware has -->
+      <!-- instructions stored in custom_metadata -->
+      <%  if asset.custom_metadatum_collection.present?  %>
+        <td><%= asset.metadata['retention_instruction'] if asset.metadata.key?('retention_instruction') %></td>
+      <%  else  %>
+        <td>Not specified</td>
+      <%  end  %>
+    <!-- If retention instructions are present, display them. -->
+    <%  else  %>
+      <td><%= find_retention_instruction_from_key(asset.retention_instruction.to_sym) %></td>
+    <%  end %>
+    <!-- There shouldn't be a situation where retention instructions are there in both custom_metadata -->
+    <!-- and in labware table's retention instruction attribute -->
+  </tr>
```

Comment:
I suggest one time data migration instead of doing these checks all the time.

Reply:
Agreed. I just added this until we get the migration story https://github.com/sanger/sequencescape/issues/4093 done. We can remove these checks as part of that story?